web/src/components/software/PatternSelector.jsx

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useCallback, useEffect, useState } from "react";
+import { SearchInput } from "@patternfly/react-core";
+import { sprintf } from "sprintf-js";
+
+import { useInstallerClient } from "~/context/installer";
+import { ValidationErrors } from "~/components/core";
+import PatternGroup from "./PatternGroup";
+import PatternItem from "./PatternItem";
+import UsedSize from "./UsedSize";
+import { _ } from "~/i18n";
+
+/**
+ * @typedef {Object} Pattern
+ * @property {string} name pattern name (internal ID)
+ * @property {string} group pattern group
+ * @property {string} summary pattern name (user visible)
+ * @property {string} description long description of the pattern
+ * @property {string} order display order (string!)
+ * @property {string} icon icon name (not path or file name!)
+ * @property {number|undefined} selected who selected the pattern, undefined
```

Comment:
NP: I usually indicate optional property instead of undefined, e.g. `@property {number} [selected] - who selected ...`. We should agree a common format.